### ISSUE TITLE
Error: Failed to extract `applied_steps_count` from `_prisma_migrations` row

### ISSUE THREAD

**Issue by MurzNN** [prisma/prisma#8234]
### Bug description

After creating and applying migration to database via `migrate dev`, on next migration generation Prisma shows me the error
```
prisma migrate dev --name migration2
Environment variables loaded from .env
Prisma schema loaded from schema.prisma
Datasource "postgresql": PostgreSQL database "my_database", schema "public" at "localhost:5432"

Error: Failed to extract `applied_steps_count` from `_prisma_migrations` row.
   0: sql_migration_connector::sql_migration_persistence::list_migrations
             at migration-engine/connectors/sql-migration-connector/src/sql_migration_persistence.rs:125
   1: migration_core::api::DevDiagnostic
             at migration-engine/core/src/api.rs:89
```
On that table I see `NULL` in `applied_steps_count` column for last migration row.
If I fill it with `1` manually, next migration generates well, but will not fill his own `1` in `applied_steps_count` on deployment action too.

### How to reproduce

1. Generate and apply first migration using `prisma migrate dev --name migration1`, see that it applied successfully.
```
Your database is now in sync with your schema.
```
2. Generate and apply second migration using `prisma migrate dev --name migration2`
3. See that it fails with error:
```
Failed to extract `applied_steps_count` from `_prisma_migrations` row.
```
4. Lookup the "_prisma_migrations" and see that  `applied_steps_count` for migration "migration1" is NULL
 
The `logs` and `rolled_back_at` are NULL too.

### Expected behavior

Value "1" for `applied_steps_count` in `_prisma_migrations` table must be filled automatically, or, if something is failed, this must be shown in output.

### Environment & setup

- OS: Ubuntu
- Database: PostgreSQL
- Node.js version: v14.16.1

### Prisma Version

```
prisma               : 2.26.0
@prisma/client       : 2.26.0
Current platform     : debian-openssl-1.1.x
Query Engine         : query-engine 9b816b3aa13cc270074f172f30d6eda8a8ce867d (at node_modules/@prisma/engines/query-engine-debian-openssl-1.1.x)
Migration Engine     : migration-engine-cli 9b816b3aa13cc270074f172f30d6eda8a8ce867d (at node_modules/@prisma/engines/migration-engine-debian-openssl-1.1.x)
Introspection Engine : introspection-core 9b816b3aa13cc270074f172f30d6eda8a8ce867d (at node_modules/@prisma/engines/introspection-engine-debian-openssl-1.1.x)
Format Binary        : prisma-fmt 9b816b3aa13cc270074f172f30d6eda8a8ce867d (at node_modules/@prisma/engines/prisma-fmt-debian-openssl-1.1.x)
Default Engines Hash : 9b816b3aa13cc270074f172f30d6eda8a8ce867d
Studio               : 0.408.0
```

**Comment by MurzNN** [comment]
Seems the problem is similar like in #8204 :confused:

**Comment by tomhoule** [comment]
Mmh this is weird. The definition of the migrations table on postgres is:

```
            CREATE TABLE _prisma_migrations (
                id                      VARCHAR(36) PRIMARY KEY NOT NULL,
                checksum                VARCHAR(64) NOT NULL,
                finished_at             TIMESTAMPTZ,
                migration_name          VARCHAR(255) NOT NULL,
                logs                    TEXT,
                rolled_back_at          TIMESTAMPTZ,
                started_at              TIMESTAMPTZ NOT NULL DEFAULT now(),
                applied_steps_count     INTEGER NOT NULL DEFAULT 0
            );
```

so it should never be null. How old is your project? Maybe it's a different migrations table from an early preview version of migrate. What version of postgres are you using?

**Comment by MurzNN** [comment]
My project is started with Prisma 2.15.0 version (or maybe little earlier), and default value for `applied_steps_count` is not set. Here is current schema of table:
```
CREATE TABLE public._prisma_migrations (
    id character varying NOT NULL,
    checksum character varying,
    finished_at timestamp with time zone,
    migration_name character varying,
    logs text,
    rolled_back_at timestamp with time zone,
    started_at timestamp with time zone,
    applied_steps_count integer
);
```

**Comment by tomhoule** [comment]
I think the easiest way to fix this for you would be running `ALTER TABLE _prisma_migrations ALTER COLUMN applied_steps_count SET DEFAULT 0` directly on the database

**Comment by MurzNN** [comment]
> I think the easiest way to fix this for you would be running `ALTER TABLE _prisma_migrations ALTER COLUMN applied_steps_count SET DEFAULT 0` directly on the database

Yeah, it solves the problem, thanks! Maybe good idea is add schema validation and migration (update to match needed schema) of this table in future Prisma versions?

**Comment by janpio** [comment]
We consider this solved as it was about an upgrade from old Migrate.

**Comment by albertsj1** [comment]
Sorry to ressurrect an old post.  We had this exact error happen to us on a production Postgres RDS instance.  A migration failed to create a new table.  After finding this post, we checked the _prisma_migrations schema and applied_steps_count was allowed to be nullable and didn't have a default value set.  Running `ALTER TABLE _prisma_migrations ALTER COLUMN applied_steps_count SET DEFAULT 0` as suggested here allowed the migration to complete.
We are currently using prisma v4.11.0.  I don't know the history of the db in production except that it was created around Aug 2, 2022 and I suspect the version of prisma used at that time was v3.9.1.
I looked back at old prisma-engine code, and as far as I can tell, the schema has had NOT NULL default 0 [forever](https://github.com/prisma/prisma-engines/blob/f136734c1997b4cba292dd8e41ea4c0a768527a2/migration-engine/connectors/sql-migration-connector/src/flavour/postgres.rs#L108).

I'm curious if anyone has any ideas how our schema could have been in this state?

Thanks
